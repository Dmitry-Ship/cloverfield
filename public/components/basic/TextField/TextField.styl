@import '../../../styles/index.styl'

.textField
  size 100% 45px
  margin-bottom 30px
  padding 0 15px
  border none 3px solid color__primary--dark
  background color__primary--light
  transition all 300ms ease-in-out
  &:focus
    background white
    border none 3px solid black
