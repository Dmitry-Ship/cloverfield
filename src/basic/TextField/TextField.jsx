import React, { Component, PropTypes } from 'react';

export default class TextField extends Component {
  constructor(){
    super();
  }

  render() {
    const {
      type,
      name,
      placeholder,
      required,
      onChange,
      onFocus,
      onBlur,
      label,
      className } = this.props;
    return (
      <input
        className={`text-field ${className}`}
        type={type}
        name={name}
        placeholder={placeholder}
        required={required}
        onFocus={onFocus}
        onBlur={onBlur}
        onChange={onChange}
         />

    )
  }
}

TextField.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  required: PropTypes.bool,
}
