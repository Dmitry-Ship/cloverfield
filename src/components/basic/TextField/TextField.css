@import '../../../stylesheets/colors.css';

.textfield {
	display: block;
	margin: 0px 0px 30px 0px;
	padding: 0 15px;
	width: 100%;
	height: 45px;
	border-top: none;
	border-right: 3px solid var(--color__primary--dark);
	border-bottom: none;
	border-left: 3px solid var(--color__primary--dark);
	background: var(--color__primary--light);
	font-size: 1rem;
	transition: all 300ms ease-in-out;
}

.text-field:focus {
	border-right: 3px solid black;
	border-left: 3px solid black;
	background: white;
}
