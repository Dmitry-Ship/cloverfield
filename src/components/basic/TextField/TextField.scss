@value colors: "../../../styles/colors.scss";
@value tertiary, secondary, primary from colors;

.textField {
  width: 100%;
  height: 45px;
  padding: 0 15px;
  border: none;
  background: secondary;
  transition: all 150ms ease-in-out;
  &::-webkit-input-placeholder {
    color: white;
  }
  &:focus {
    background: primary;
    &::-webkit-input-placeholder {
      color: secondary;
    }
  }
}

.errorMessage {
  position: absolute;
  color: red;
  margin: 5px 0 0 0;
}

.wrapper {
  position: relative;
}

.topLabel {
  position: absolute;
  bottom: 100%;
  color: tertiary;
  margin: 5px 0 0 0;
}