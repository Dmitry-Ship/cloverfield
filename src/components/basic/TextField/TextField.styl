@import '../../../styles/colors.styl'

.textField
  display block
  size 100% 45px
  padding 0 15px
  margin 0px 0px 30px 0px
  border-top none
  border-bottom none
  border-left 3px solid $color__primary--dark
  border-right 3px solid $color__primary--dark
  background $color__primary--light
  transition all 300ms ease-in-out
  &:focus
    background white
    border-left 3px solid black
    border-right 3px solid black
