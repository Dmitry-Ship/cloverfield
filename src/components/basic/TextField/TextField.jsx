import PropTypes from 'prop-types';
import React from 'react';
import { wrapper, textField, errorMessage, topLabel } from './TextField.scss';

const TextField = ({ label, className, error, ...rest }) => (
  <div className={`${wrapper} ${className}`}>
    {label && <label className={topLabel} >{label}</label>}
    <input className={textField} {...rest} />
    {error && <p className={errorMessage}>{error}</p>}
  </div>
);

export default TextField;

TextField.defaultProps = {
  className: '',
  type: 'text',
  placeholder: 'Placeholder',
  label: null,
  error: null,
};

TextField.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf(['text', 'email', 'number', 'password', 'date', 'time']),
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.string,
};
