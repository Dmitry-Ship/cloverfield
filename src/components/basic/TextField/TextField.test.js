import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextField from './TextField';

describe('<TextField />', () => {
  const onChange = jest.fn();
  it('should render correctly', () => {
    const tree = renderer.create(
      <TextField value="" onChange={onChange} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call expandImage', () => {
    const wrapper = shallow(
      <TextField value="" onChange={onChange} />
    );

    wrapper.find('input').first().simulate('change', { target: { value: 'My new value' } });

    expect(onChange).toBeCalledWith({ target: { value: 'My new value' } });
  });
});
