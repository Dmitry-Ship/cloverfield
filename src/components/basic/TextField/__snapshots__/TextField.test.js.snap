// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render correctly 1`] = `
<div
  className="wrapper "
>
  <input
    className="textField"
    onChange={[Function]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
</div>
`;
